kalliope/core/Models/Signal.py

```diff
@@ -0,0 +1,51 @@
+class Signal(object):
+    """
+    This Class is representing a Signal which is corresponding to an input action that should start executing neuron
+    list when triggered
+    """
+
+    def __init__(self, name=None, parameters=None):
+        self.name = name
+        self.parameters = parameters
+
+    def serialize(self):
+        """
+        This method allows to serialize in a proper way this object
+
+        :return: A dict of name and parameters
+        :rtype: Dict
+        """
+        return {
+            'name': self.name,
+            'parameters': self.parameters
+        }
+
+    def __str__(self):
+        """
+        Return a string that describe the signal. If a parameter contains the word "password",
+        the output of this parameter will be masked in order to not appears in clean in the console
+        :return: string description of the neuron
+        """
+        returned_dict = {
+            'name': self.name,
+            'parameters': self.parameters
+        }
+
+        cleaned_parameters = dict()
+        if isinstance(self.parameters, dict):
+            for key, value in self.parameters.items():
```

Comment:
Refactor the for loop. Use instead : 
returned_dict = self.parameters
if self.parameters["password"] != None
    returned_dict["password"] = "******"

Reply:
actually here we are not looking exactly for a key called "password" but for each key that contain the work "password".